### ISSUE TITLE
install_github fails with unzip error

### ISSUE THREAD

**Issue by plantarum** [hadley/devtools#1464]
I'm trying to install my own package from github: plantarum/flowPloidy, onto a Windows computer. Starting from a fresh session, installing devtools from Cran, I try:

    install_github("plantarum/flowPloidy", dependencies = TRUE)

This fails with:

```
Error in utils::unzip(src, exdir = target) : 
  cannot open file 'C:/Users/jamest/AppData/Local/Temp/RtmpKWeuxp/devtools1fc45f5c7551/plantarum-flowPloidy-88ad2c0/R/FlowHist.R': No such file or directory
```

I looked into those directories, and the toplevel directory 'C:/Users/jamest/AppData/Local/Temp/RtmpKWeuxp/devtools1fc45f5c7551/plantarum-flowPloidy-88ad2c0/ is there, and in the downloaded zip file the R folder exists and contains all the code files as you expect. 

I tried debugging my way through the code, and got as far as the function get0 before I gave up trying to figure out the ultimate source of the error.

install_github works no problem for this package under Linux.

Thanks for any help or suggestions,

Tyler

R version 3.3.2 (2016-10-31) -- "Sincere Pumpkin Patch"

**Comment by plantarum** [comment]
I just tried installing the zip file directly and got the same problem, so the problem is probably not with devtools itself?

```
install.packages("C:/Users/jamest/Downloads/flowPloidy-master.zip", repos = NULL, type = "win.binary")
Error in install.packages : cannot open file 'C:/Users/jamest/Documents/R-3.3.2/library/file4e4c8cc0b/flowPloidy-master/R/FlowHist.R': No such file or directory
```

**Comment by jianlingfan** [comment]
I have the same problem for several weeks too, still looking for a solution.

**Comment by alex23lemm** [comment]
Today, I ran into the problem by letting several people install the `tutor` package using `devtools`. All Mac and Linux users had no issues and also the installation for the majority of the Windows users went well. 

However, some of the Windows users ran into the problem described above which looked as follows in our case:
 ```r
> devtools::install_github("rstudio/tutor")
Downloading GitHub repo rstudio/tutor@master
from URL https://api.github.com/repos/rstudio/tutor/zipball/master
Error in utils::unzip(src, exdir = target) : 
  cannot open file 'C:/Users/mshumper/AppData/Local/Temp/RtmpeE4D2Q/devtools23e061002ad9/rstudio-tutor-1476bf6/R/ace.R': No such file or directory
```
`All users installed the current R version 3.3.3 and the current RStudio version 1.0.136  today prior to trying to install `tutor` via `devtools`.

Even though I am on Windows as well I was not able to reproduce the bug on my machine.

**Comment by dfifield** [comment]
This sounds very similar to the problem described in Issue #1433. The work-around was to get rid of .Rbuildignore.

**Comment by hadley** [comment]
@gaborcsardi @jimhester have you seen this before?

**Comment by gaborcsardi** [comment]
Haven't. Can't reproduce. Maybe a "wrong" unzip program again? Although I am not sure how removing `.Rbuildignore` would fix that....

**Comment by imanuelcostigan** [comment]
This looks like the same bug that has been bedevilling my team for years (!). We all have Windows 7 x64 PC, each with the same corporate "build" and have same admin privileges. Yet `install_github()` works on my PC, but not on any other person's PC. Everyone else has this same issue. See this SO thread: https://stackoverflow.com/questions/41293748/unzip-fails-using-install-github/41293968

In particular, note this behaviour:

>  when they download the zip directly from Github's website and try to extract into their user folder, they get the same behaviour from Winzip, but 7-zip can successfully unpack the file.

I've spent quite a bit of time debugging this all to no avail. Squashing this bug would be massive for us.

**Comment by gaborcsardi** [comment]
@imanuelcostigan Can you pls share a zip file that is problematic, and also the winzip version your colleagues have?

**Comment by hadley** [comment]
@imanuelcostigan you might also try comparing `getOption("unzip")` on your computer vs. your colleagues

**Comment by hadley** [comment]
It's possible that this may also be related to the path length.

**Comment by gaborcsardi** [comment]
@hadley In the original error, the path is only about 90 characters long, that should be ok.

**Comment by hadley** [comment]
@gaborcsardi good point. I'll keep the other issue open

**Comment by imanuelcostigan** [comment]
@hadley getOption("unzip") yields "internal" for me.

@pgwatson @jonathanrandall can you please add what you get from the same command to this thread? 

Thanks

**Comment by jonathanrandall** [comment]
"internal"

**Comment by pgwatson** [comment]
getOption(“unzip”) also yields “internal” and fails for me

However if I set the unzip option to point to the unzip.exe program bundled with the github application then everything appears to work correctly.

The version info for unzip.exe bundled with github is

[cid:image001.png@01D30C46.4C98E950]

Cheers

Paul Watson
Senior Manager, Model Risk, Risk Analytics & Insights
T +61 2 8254 4125

**Comment by pgwatson** [comment]
Interestingly if I point instead to the version 6.00 unzip bundled with Rtools then the devtools::install_github process still fails

The Rtools version of unzip (which doesn’t work) has several different compilation flags set compared to the version from github (which does work)

Compilation flags follow:
Rtools/unzip on the left and github/unzip on the right

[cid:image002.png@01D30C52.C7B3D760] [cid:image001.png@01D30C52.2524B2D0]

Regards

Paul Watson
Senior Manager, Model Risk, Risk Analytics & Insights
T +61 2 8254 4125

From: Watson, Paul
Sent: Thursday, 3 August 2017 10:56 AM
To: 'hadley/devtools'; hadley/devtools
Cc: Mention
Subject: RE: [hadley/devtools] install_github fails with unzip error (#1464)

getOption(“unzip”) also yields “internal” and fails for me

However if I set the unzip option to point to the unzip.exe program bundled with the github application then everything appears to work correctly.

The version info for unzip.exe bundled with github is

[cid:image001.png@01D30C52.2524B2D0]

Cheers

Paul Watson
Senior Manager, Model Risk, Risk Analytics & Insights
T +61 2 8254 4125

**Comment by hadley** [comment]
I'm going to reopen this so we can include some basic fix for devtools (to at least help diagnose the problem)

**Comment by imanuelcostigan** [comment]
@hadley let me know if my team or I can help provide you any more info.

**Comment by jimhester** [comment]
I am pretty sure this issue is only tangentially related to devtools code, the error is coming from the internal unzip code. 

https://github.com/wch/r-source/blob/780021752eb83a71e2198019acf069ba8741103b/src/main/dounzip.c#L177

Possible causes could be 
- lack of write permissions in `tempdir()` or one of the children for the user R is running under
- `tempdir()` or one of the child directories being removed. This could be done automatically by [certain windows options](https://www.laptopmag.com/articles/how-to-use-windows-10-storage-sense).

**Comment by hadley** [comment]
And devtools does respect the `unzip` option, so you might able to set the option to overcome your problems (but you might want to check that different versions of zip consistently fail or not in case the problem has a stochastic component).

**Comment by hadley** [comment]
I think we'll close this issue since it doesn't seem like devtools' problem - but feel free to continue to contribute insights here (so people find them when googling), and we are likely to explore alternatives to the internal zip utility in remotes.

**Comment by imanuelcostigan** [comment]
@jimhester I'm pretty sure that the issue isn't caused by either of those. We continued to see the same error when setting a different temp directory that users definitely had read/write access to and the directory was not being removed as we were able to see the directory after the process failed. In fact, unzipping the package failed using WinZip with the same sort of error in a user directory. 

In summary, what we have found was the unzipping the package:

* worked for everyone using 7-zip
* worked for everyone using Github compiled `unzip` (but per comment above, GH's unzip is compiled differently to the one bundled by R, even though they are the same version)
* worked for me using WinZip but no one else
* worked for me using R bundled unzip code, but not for anyone else in my team

**Comment by imanuelcostigan** [comment]
`install_github()` has now stopped working for me on my new machine. Following @pgwatson [suggestion above](https://github.com/hadley/devtools/issues/1464#issuecomment-319849807), here's a work around: 

* [Download and install Git](https://git-scm.com). 
* Set the `unzip` option to version of `unzip.exe` supplied by Git e.g. `options(unzip = "c:/Program Files/Git/usr/bin/unzip.exe")`. You can obviously set this option at startup by adding this to your `.Rprofile`.

Everything now works.
